upt_macports/upt_macports.py

```diff
@@ -176,6 +180,47 @@ def create_package(self, upt_pkg, output=None):
             pkg_cls = pkg_classes[upt_pkg.frontend]
         except KeyError:
             raise upt.UnhandledFrontendError(self.name, upt_pkg.frontend)
-
         packager = pkg_cls()
         packager.create_package(upt_pkg, output)
+
+    def pkg_exist(self):
+        prefix = {
+            'pypi': 'py-',
+            'rubygems': 'rb-',
+            'cpan': 'p5-',
+        }
+        port_name = prefix.get(self.upt_pkg.frontend, '') + self.upt_pkg.name
+        self.logger.info(f'Checking MacPorts tree for port {port_name}')
+        cmd = f'port info --version {port_name}'
+        port = subprocess.getoutput(cmd)
+        self.logger.info(f'Current Upstream version is {self.upt_pkg.version}')
+        if port[:5] == 'Error':
+            self.logger.info(f'{port_name} not found in MacPorts tree')
+            self.curr_ver = ''
+            return False
+        else:
+            self.curr_ver = port.split()[1]
+            self.logger.info(
+                f'Current MacPorts Version for {port_name} is {self.curr_ver}')
+            return True
+
+    def check_needed(self, upt_pkg, constraint):
```

Comment:
This whole method is difficult to read, in my opinion.

Don't do:

```python
if X:
    # Some
    # Long
    # Code
    # For
    # The
    # Most
    # Common
    # Case
else:
    return Y```

Instead, write:

```python
if not X:
    return Y

# Some
# Long
# Code
# ...```

This way the reader does not have to remember as much context when reading the "real" code.